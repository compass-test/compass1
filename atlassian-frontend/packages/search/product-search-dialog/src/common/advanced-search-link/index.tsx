export { AdvancedSearchLink } from './advanced-search-link';
export {
  AdvancedSearchContent,
  CenteredAdvancedSearchGroup,
  StyledAdvancedSearch,
} from './advanced-search-link.styled';

export { default as MultiSiteAdvancedSearchLink } from './multi-site-advanced-search-link';
