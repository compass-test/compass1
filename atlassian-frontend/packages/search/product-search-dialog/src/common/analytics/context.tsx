import {
  ContextPayload,
  NonPrivacySafeContext,
  SearchResultContextPayload,
} from '../../common/analytics/events';
import React, { Reducer, useCallback, useReducer } from 'react';
import { getWordCount, sha1Hash } from '../../common/analytics/utils';

interface AdditionalAnalyticsContextState {
  queryVersion: number;
  isLoading: boolean;
}

export type AnalyticsContextAction =
  | { type: 'reset' }
  | {
      type: 'queryVersion';
      value: number;
    }
  | {
      type: 'isLoading';
      value: boolean;
    };

const defaultAnalyticsContextState: AdditionalAnalyticsContextState = {
  queryVersion: 0,
  isLoading: false,
};

export type SupportedContext = SearchResultContextPayload | ContextPayload;

const analyticsContextReducer: Reducer<
  AdditionalAnalyticsContextState,
  AnalyticsContextAction
> = (state, action) => {
  switch (action.type) {
    case 'queryVersion':
      return { ...state, queryVersion: action.value };
    case 'isLoading':
      return { ...state, isLoading: action.value };
    case 'reset':
      return defaultAnalyticsContextState;
    default:
      throw new Error();
  }
};

type AnalyticsHook = (
  query: string,
) => {
  isLoading: boolean;
  setAdditionalAnalyticsContext: React.Dispatch<AnalyticsContextAction>;
  addAnalyticContext: () => ContextPayload;
  nonPrivacySafeContext: () => NonPrivacySafeContext;
  queryVersion: number;
};

export const useAnalyticsContext: AnalyticsHook = (query: string) => {
  const [
    { queryVersion, isLoading },
    setAdditionalAnalyticsContext,
  ] = useReducer(analyticsContextReducer, defaultAnalyticsContextState);
  const addAnalyticContext = useCallback(() => {
    return {
      wordCount: getWordCount(query),
      queryLength: query.length,
      queryHash: sha1Hash(query),
      queryVersion: queryVersion,
    };
  }, [query, queryVersion]);
  const nonPrivacySafeContext = useCallback(() => {
    return { query };
  }, [query]);

  return {
    isLoading,
    setAdditionalAnalyticsContext,
    addAnalyticContext,
    nonPrivacySafeContext,
    queryVersion,
  };
};
