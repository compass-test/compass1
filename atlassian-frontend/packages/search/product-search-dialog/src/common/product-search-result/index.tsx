export { default as ProductSearchResult } from './product-search-result';
