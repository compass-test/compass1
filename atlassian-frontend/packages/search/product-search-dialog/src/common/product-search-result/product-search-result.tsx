import React, { FunctionComponent, useCallback } from 'react';
import { SearchResult, SearchResultProps } from '@atlassian/search-dialog';
import {
  SearchDialogAnalyticsContext,
  useAnalytics,
  ScreenType,
  NonPrivacySafeContext,
  SearchResultContextPayload,
  onSearchResultHighlighted,
  onSearchResultSelected,
} from '../analytics';
import { useSessionUserInput } from '../../extensible/user-input-provider';

interface PropsWithAnalyticsContext extends SearchResultProps {
  analyticContext: SearchResultContextPayload;
  nonPrivacySafeAnalyticContext?: NonPrivacySafeContext;
  screen: ScreenType;
}

interface PropsWithAnalytics extends SearchResultProps {
  screen: ScreenType;
}

const SearchResultWithAnalytics = ({
  screen,
  onSelect,
  onHighlighted,
  ...rest
}: PropsWithAnalytics) => {
  const { fireAnalyticsEvent } = useAnalytics();
  const {
    stickySearchEnabled: isSticky,
    isStickyUpdated,
  } = useSessionUserInput();

  const onHighlightCallback = useCallback(() => {
    fireAnalyticsEvent(
      onSearchResultHighlighted({
        screen,
      }),
    );

    onHighlighted?.();
  }, [fireAnalyticsEvent, onHighlighted, screen]);

  const onSelectCallback = useCallback(
    (e: React.MouseEvent<Element, MouseEvent> | KeyboardEvent) => {
      fireAnalyticsEvent(
        onSearchResultSelected({
          screen,
          isSticky,
          isStickyUpdated,
        }),
      );

      onSelect?.(e);
    },
    [fireAnalyticsEvent, isSticky, isStickyUpdated, onSelect, screen],
  );

  return (
    <SearchResult
      {...rest}
      onSelect={onSelectCallback}
      onHighlighted={onHighlightCallback}
    />
  );
};

const ProductSearchResult: FunctionComponent<PropsWithAnalyticsContext> = ({
  analyticContext,
  nonPrivacySafeAnalyticContext,
  ...rest
}) => {
  return (
    <SearchDialogAnalyticsContext
      analyticContext={analyticContext}
      nonPrivacySafeAnalyticContext={nonPrivacySafeAnalyticContext || {}}
    >
      <SearchResultWithAnalytics {...rest} />
    </SearchDialogAnalyticsContext>
  );
};

export default ProductSearchResult;
