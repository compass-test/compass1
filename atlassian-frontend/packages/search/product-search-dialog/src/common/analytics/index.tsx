import * as ContextImports from './context';
export type SupportedContext = ContextImports.SupportedContext;
export type AnalyticsContextAction = ContextImports.AnalyticsContextAction;
export { SearchDialogAnalyticsContext } from './search-dialog-analytics-context';
export { SearchResultsShownHandler } from './results-shown-handler';
export { withAnalytics } from './with-analytics';
export { useAnalytics } from './use-analytics';
export { DialogDismissedHandler } from './dialog-dismissed-handler';
export { TextEnteredHandler } from './text-entered-handler';
export { useAnalyticsContext } from './context';
export { sha1Hash, getTrigger, isNewTab, getWordCount } from './utils';
export {
  onFilterUnselect,
  onFilterSelect,
  onAdvancedSearchSelected,
  onMultiSiteAdvancedSearchSelected,
  onDismissed,
  onExperimentLoaded,
  onFiltersCleared,
  onMoreFiltersSelected,
  onPaneLoadComplete,
  onPreQueryEmpty,
  onPreQueryScreenViewed,
  onRequestMade,
  onSearchResultHighlighted,
  onSearchResultSelected,
  onSearchResultsShown,
  onShowAllClicked,
  onShowMoreClicked,
  onTabSelected,
  onTextEntered,
  AdvancedSearchLinkSubjectId,
  selectEvent,
} from './events';

export type {
  ScreenType,
  NonPrivacySafeContext,
  OnPreQueryScreenViewedProps,
  OnSearchResultHighlightedProps,
  OnSearchResultSelectedProps,
  OnShowAllClickedProps,
  OnShowMoreClickedProps,
  SearchResultContextPayload,
  SectionID,
  ResultAnalyticData,
  ResultSectionAnalyticData,
} from './events';
