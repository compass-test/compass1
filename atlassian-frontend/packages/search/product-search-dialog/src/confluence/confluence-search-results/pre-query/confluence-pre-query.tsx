import { LinkComponent, SearchResultSection } from '@atlassian/search-dialog';
import React from 'react';
import { FormattedMessage } from 'react-intl';
import {
  ConfItemResults,
  ConfPeopleResults,
  ConfSpaceResults,
  withClients,
} from '../../clients';
import {
  withAnalytics,
  SearchResultsShownHandler,
} from '../../../common/analytics';
import {
  onAdvancedSearchSelected,
  AdvancedSearchAnalyticsProps,
} from '../../../common/analytics/events';
import { messages } from '../../../messages';
import { ResultAvatar as Avatar } from '../../../common/result-avatar';
import { injectSearchSession } from '../../../common/search-session-provider';
import { addQuery } from '../../../utils/url-utils';
import { ProductSearchResult } from '../../../common/product-search-result';
import { LoadingSpinner } from '../../../common/loading-spinner';
import {
  getResultsDetailsAnalyticData,
  getAvatarForConfluenceObjectResult,
  firePeopleSearchSelected,
} from '../confluence-search-results-utils';
import ConfluenceSearchResultSectionLink from '../confluence-search-result-section-link';

interface Props {
  items: {
    isLoading: boolean;
    results: ConfItemResults | null;
  };
  spaces: {
    isLoading: boolean;
    results: ConfSpaceResults | null;
  };
  people: {
    isLoading: boolean;
    results: ConfPeopleResults | null;
  };
  searchSessionId: string;
  siteUrl?: string;
  advancedSearchSelected: (props: AdvancedSearchAnalyticsProps) => any;
  linkComponent?: LinkComponent;
  formatDate?: (lastModified: string) => JSX.Element;
  isLoading: boolean;
  isCollapsed?: boolean;
}

const PRE_QUERY_ITEM_LIMIT = 10;
const PRE_QUERY_SPACE_LIMIT = 3;
const PRE_QUERY_PEOPLE_LIMIT = 3;

// We disable prefer-stateless-function until confluence-post-query is also a function component. Easier to compare the 2 this way.
// eslint-disable-next-line react/prefer-stateless-function
export class ConfluencePreQuery extends React.Component<Props> {
  render() {
    const {
      items,
      spaces,
      people,
      linkComponent,
      formatDate,
      searchSessionId,
      isLoading,
      advancedSearchSelected,
      isCollapsed,
      siteUrl = '',
    } = this.props;

    let currentSectionIndex = 0;
    let totalNumberOfItemsInPreviousSections = 0;
    const avatarSize = isCollapsed ? 'small' : 'xsmall';

    const itemResults =
      items.results?.items.slice(0, PRE_QUERY_ITEM_LIMIT) || [];
    const itemSection = itemResults.length ? (
      <SearchResultSection
        title={
          <FormattedMessage
            {...messages.common_recently_viewed_section_heading}
          />
        }
      >
        {itemResults.map((result, index) => (
          <ProductSearchResult
            key={`item_${result.resultId}`}
            title={result.name}
            href={addQuery(result.href, 'search_id', searchSessionId)}
            icon={getAvatarForConfluenceObjectResult(
              result.contentType,
              result.name,
              isCollapsed,
            )}
            containerDetail={result.containerName}
            timeDetail={
              formatDate && result.lastModified
                ? formatDate(result.lastModified)
                : undefined
            }
            linkComponent={linkComponent}
            screen="preQuerySearchResults"
            isCollapsed={isCollapsed}
            analyticContext={{
              sectionIndex: currentSectionIndex,
              sectionId: 'confluence-item',
              containerId: result.containerId,
              globalIndex: totalNumberOfItemsInPreviousSections + index,
              indexWithinSection: index,
              isCached: result.isRecentResult,
            }}
          />
        ))}
      </SearchResultSection>
    ) : null;

    currentSectionIndex = itemSection
      ? currentSectionIndex + 1
      : currentSectionIndex;
    totalNumberOfItemsInPreviousSections += itemResults.length;

    const spaceResults =
      spaces.results?.items.slice(0, PRE_QUERY_SPACE_LIMIT) || [];
    const spaceSection = spaceResults.length ? (
      <SearchResultSection
        title={
          <FormattedMessage
            {...messages.common_recently_spaces_section_heading}
          />
        }
      >
        {spaceResults.map((result, index) => (
          <ProductSearchResult
            key={`spaces_${result.resultId}`}
            title={result.name}
            href={addQuery(result.href, 'search_id', searchSessionId)}
            icon={
              <Avatar
                borderColor="transparent"
                src={result.avatarUrl}
                appearance="square"
                size={avatarSize}
              />
            }
            linkComponent={linkComponent}
            screen="preQuerySearchResults"
            analyticContext={{
              sectionIndex: currentSectionIndex,
              sectionId: 'confluence-space',
              containerId: null,
              globalIndex: totalNumberOfItemsInPreviousSections + index,
              indexWithinSection: index,
              isCached: true,
            }}
          />
        ))}
      </SearchResultSection>
    ) : null;

    currentSectionIndex = spaceSection
      ? currentSectionIndex + 1
      : currentSectionIndex;
    totalNumberOfItemsInPreviousSections += spaceResults.length;

    const peopleResults =
      people.results?.items.slice(0, PRE_QUERY_PEOPLE_LIMIT) || [];
    let peopleSection: JSX.Element | null = null;

    if (!isLoading && people.isLoading) {
      peopleSection = <LoadingSpinner />;
    } else {
      if (peopleResults.length > 0) {
        peopleSection = (
          <SearchResultSection
            title={
              <FormattedMessage
                {...messages.common_recently_worked_with_section_heading}
              />
            }
          >
            {peopleResults.map((result, index) => (
              <ProductSearchResult
                key={`people_${result.resultId}`}
                title={result.name}
                href={addQuery(result.href, 'search_id', searchSessionId)}
                icon={
                  <Avatar
                    borderColor="transparent"
                    src={result.avatarUrl}
                    appearance="circle"
                    size={avatarSize}
                  />
                }
                containerDetail={
                  result.mentionName ? `@${result.mentionName}` : ''
                }
                linkComponent={linkComponent}
                screen="preQuerySearchResults"
                analyticContext={{
                  sectionIndex: currentSectionIndex,
                  sectionId: 'confluence-people',
                  containerId: null,
                  globalIndex: totalNumberOfItemsInPreviousSections + index,
                  indexWithinSection: index,
                  isCached: true,
                }}
              />
            ))}
            <ConfluenceSearchResultSectionLink
              linkComponent={linkComponent}
              href={`${siteUrl}/wiki/people`}
              id="people-search-link"
            >
              <span
                role="none"
                onClick={(e) =>
                  firePeopleSearchSelected(e, advancedSearchSelected)
                }
              >
                <FormattedMessage {...messages.common_search_people_link} />
              </span>
            </ConfluenceSearchResultSectionLink>
          </SearchResultSection>
        );
      }
    }

    currentSectionIndex = peopleSection
      ? currentSectionIndex + 1
      : currentSectionIndex;
    totalNumberOfItemsInPreviousSections += peopleResults.length;

    return (
      <span data-test-id="search-dialog-confluence-pre-query">
        {itemSection}
        {spaceSection}
        {peopleSection}
        <SearchResultsShownHandler
          isPreQuery
          isLoading={isLoading}
          resultCount={totalNumberOfItemsInPreviousSections}
          sectionCount={currentSectionIndex}
          results={getResultsDetailsAnalyticData(
            itemResults,
            spaceResults,
            peopleResults,
            true,
          )}
        />
      </span>
    );
  }
}

export default withAnalytics({
  advancedSearchSelected: onAdvancedSearchSelected,
})(withClients(injectSearchSession(ConfluencePreQuery)));
