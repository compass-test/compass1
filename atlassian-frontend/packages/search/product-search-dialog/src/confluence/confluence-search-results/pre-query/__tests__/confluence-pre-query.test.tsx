import React from 'react';
import { shallow } from 'enzyme';
import { assertSearchResultSectionsHaveTitles } from '../../../../__tests__/__fixtures__/intl-test-helpers';
import {
  createPageBlogAttachmentResults,
  createSpaceResponse,
  createPeopleResults,
  enableDeterministicResponses,
} from '../../../../__tests__/__fixtures__/mock-search-results';
import { ConfluencePreQuery } from '../confluence-pre-query';
import { SearchResultSection } from '@atlassian/search-dialog';
import { messages } from '../../../../messages';
import { ProductSearchResult } from '../../../../common/product-search-result';
import { SearchResultsShownHandler } from '../../../../common/analytics';

jest.mock('@atlassian/search-dialog', () => ({
  SearchResultSection: (props: any) => <div {...props} />,
  SearchResult: () => <div />,
  SearchResultSectionLink: () => <div />,
}));

jest.mock('../../../confluence-features', () => ({
  useFeatures: () => {},
}));

jest.mock('../../../../common/product-search-result', () => ({
  ProductSearchResult: () => <div />,
}));

describe('<ConfluencePreQuery />', () => {
  const createResults = () => ({
    items: {
      isLoading: false,
      results: createPageBlogAttachmentResults(3),
    },
    spaces: {
      isLoading: false,
      results: createSpaceResponse(3),
    },
    people: {
      isLoading: false,
      results: createPeopleResults(3),
    },
  });

  const advancedSearchSelected = jest.fn();
  const searchSessionId = 'testSearchSessionId';

  const baseProps = {
    advancedSearchSelected,
    searchSessionId,
    isLoading: false,
  };

  const emptyProps = {
    items: {
      isLoading: false,
      results: {
        items: [],
        totalSize: 0,
        timings: 0,
      },
    },
    spaces: {
      isLoading: false,
      results: {
        items: [],
        timings: 0,
      },
    },
    people: {
      isLoading: false,
      results: {
        items: [],
        timings: 0,
      },
    },
  };

  beforeAll(() => {
    enableDeterministicResponses();
  });

  it('render with results matches snapshot', () => {
    const wrapper = shallow(
      <ConfluencePreQuery {...baseProps} {...createResults()} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('render with results and siteurl , search people has proper URL', () => {
    const wrapper = shallow(
      <ConfluencePreQuery
        {...baseProps}
        siteUrl="TEST_URL"
        {...createResults()}
      />,
    );

    expect(wrapper.find('#people-search-link').first().props().href).toEqual(
      'TEST_URL/wiki/people',
    );
  });

  describe('with 20 results', () => {
    it('render no more than 10 item results', () => {
      const wrapper = shallow(
        <ConfluencePreQuery
          {...baseProps}
          {...emptyProps}
          items={{
            isLoading: false,
            results: createPageBlogAttachmentResults(20),
          }}
        />,
      );

      expect(wrapper.find(ProductSearchResult)).toHaveLength(10);
    });

    it('fires search results shown with 10 results and 1 section', () => {
      const wrapper = shallow(
        <ConfluencePreQuery
          {...baseProps}
          {...emptyProps}
          items={{
            isLoading: false,
            results: createPageBlogAttachmentResults(20),
          }}
        />,
      );

      expect(wrapper.find(SearchResultsShownHandler).prop('resultCount')).toBe(
        10,
      );
      expect(wrapper.find(SearchResultsShownHandler).prop('sectionCount')).toBe(
        1,
      );
    });
  });

  it('render no more than 3 people results', () => {
    const wrapper = shallow(
      <ConfluencePreQuery
        {...baseProps}
        {...emptyProps}
        people={{ isLoading: false, results: createPeopleResults(20) }}
      />,
    );
    expect(wrapper.find(ProductSearchResult)).toHaveLength(3);
  });

  it('render no more than 3 space results', () => {
    const wrapper = shallow(
      <ConfluencePreQuery
        {...baseProps}
        {...emptyProps}
        spaces={{ isLoading: false, results: createSpaceResponse(20) }}
      />,
    );
    expect(wrapper.find(ProductSearchResult)).toHaveLength(3);
  });

  it('does not render empty space results', () => {
    const wrapper = shallow(
      <ConfluencePreQuery
        {...baseProps}
        {...createResults()}
        spaces={{ isLoading: false, results: { items: [], timings: 0 } }}
      />,
    );

    const sections = wrapper.find(SearchResultSection);

    expect(sections).toHaveLength(2);
    assertSearchResultSectionsHaveTitles(sections, [
      messages.common_recently_viewed_section_heading,
      messages.common_recently_worked_with_section_heading,
    ]);
  });

  it('does not render empty item results', () => {
    const wrapper = shallow(
      <ConfluencePreQuery
        {...baseProps}
        {...createResults()}
        items={{
          isLoading: false,
          results: { items: [], totalSize: 0, timings: 0 },
        }}
      />,
    );

    const sections = wrapper.find(SearchResultSection);

    expect(sections).toHaveLength(2);
    assertSearchResultSectionsHaveTitles(sections, [
      messages.common_recently_spaces_section_heading,
      messages.common_recently_worked_with_section_heading,
    ]);
  });

  it('does not render empty people results', () => {
    const wrapper = shallow(
      <ConfluencePreQuery
        {...baseProps}
        {...createResults()}
        people={{ isLoading: false, results: { items: [], timings: 0 } }}
      />,
    );

    const sections = wrapper.find(SearchResultSection);

    expect(sections).toHaveLength(2);
    assertSearchResultSectionsHaveTitles(sections, [
      messages.common_recently_viewed_section_heading,
      messages.common_recently_spaces_section_heading,
    ]);
  });

  it('renders no results screen if no results at all', () => {
    const wrapper = shallow(
      <ConfluencePreQuery {...baseProps} {...emptyProps} />,
    );

    const sections = wrapper.find(SearchResultSection);

    expect(sections).toHaveLength(0);
  });

  it('fires analytics event on people search selected without modifiers', () => {
    const wrapper = shallow(
      <ConfluencePreQuery {...baseProps} {...createResults()} />,
    );

    const peopleSearchLink = wrapper.find('#people-search-link').find('span');
    peopleSearchLink.simulate('click', { screenX: 30 });

    expect(advancedSearchSelected).toHaveBeenCalledWith({
      trigger: 'click',
      actionSubjectId: 'confluencePeopleSearchLink',
      isLoading: false,
      newTab: false,
    });
  });

  ['ctrlKey', 'shiftKey', 'metaKey'].forEach((modifier) => {
    it(`fires analytics event on people search selected with ${modifier}`, () => {
      const wrapper = shallow(
        <ConfluencePreQuery {...baseProps} {...createResults()} />,
      );
      const peopleSearchLink = wrapper.find('#people-search-link').find('span');

      peopleSearchLink.simulate('click', { screenX: 30, [modifier]: true });

      expect(advancedSearchSelected).toHaveBeenCalledWith({
        trigger: 'click',
        actionSubjectId: 'confluencePeopleSearchLink',
        isLoading: false,
        newTab: true,
      });
    });
  });
});
