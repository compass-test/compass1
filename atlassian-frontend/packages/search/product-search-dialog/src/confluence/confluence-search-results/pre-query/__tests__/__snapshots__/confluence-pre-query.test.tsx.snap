// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfluencePreQuery /> render with results matches snapshot 1`] = `
<span
  data-test-id="search-dialog-confluence-pre-query"
>
  <SearchResultSection
    title={
      <FormattedMessage
        defaultMessage="Recently viewed"
        description=""
        id="product_search_dialog.recently_viewed_section_heading"
        values={Object {}}
      />
    }
  >
    <ProductSearchResult
      analyticContext={
        Object {
          "containerId": "UNAVAILABLE",
          "globalIndex": 0,
          "indexWithinSection": 0,
          "isCached": false,
          "sectionId": "confluence-item",
          "sectionIndex": 0,
        }
      }
      containerDetail="Hilpert Group"
      href="http://baseUrl/spaces/a430a73d-9e9b-4cce-86a5-1618965ffea6?search_id=testSearchSessionId"
      icon={
        <Page16Icon
          label="soluta non omnis vel ut"
          primaryColor="#2684FF"
        />
      }
      key="item_a430a73d-9e9b-4cce-86a5-1618965ffea6"
      screen="preQuerySearchResults"
      title="soluta non omnis vel ut"
    />
    <ProductSearchResult
      analyticContext={
        Object {
          "containerId": "UNAVAILABLE",
          "globalIndex": 1,
          "indexWithinSection": 1,
          "isCached": false,
          "sectionId": "confluence-item",
          "sectionIndex": 0,
        }
      }
      containerDetail="Haag - Balistreri"
      href="http://baseUrl/spaces/eec8184c-94bf-478c-9398-f8e79c41fa45?search_id=testSearchSessionId"
      icon={<React.Fragment />}
      key="item_eec8184c-94bf-478c-9398-f8e79c41fa45"
      screen="preQuerySearchResults"
      title="nisi.jpg"
    />
    <ProductSearchResult
      analyticContext={
        Object {
          "containerId": "UNAVAILABLE",
          "globalIndex": 2,
          "indexWithinSection": 2,
          "isCached": false,
          "sectionId": "confluence-item",
          "sectionIndex": 0,
        }
      }
      containerDetail="Ullrich - Stokes"
      href="http://baseUrl/spaces/0e05cef4-042c-4adc-ad48-06618d01ced8?search_id=testSearchSessionId"
      icon={
        <Page16Icon
          label="magni fugiat quis et voluptas"
          primaryColor="#2684FF"
        />
      }
      key="item_0e05cef4-042c-4adc-ad48-06618d01ced8"
      screen="preQuerySearchResults"
      title="magni fugiat quis et voluptas"
    />
  </SearchResultSection>
  <SearchResultSection
    title={
      <FormattedMessage
        defaultMessage="Recent spaces"
        description=""
        id="product_search_dialog.recent_spaces_section_heading"
        values={Object {}}
      />
    }
  >
    <ProductSearchResult
      analyticContext={
        Object {
          "containerId": null,
          "globalIndex": 3,
          "indexWithinSection": 0,
          "isCached": true,
          "sectionId": "confluence-space",
          "sectionIndex": 1,
        }
      }
      href="http://baseUrl/spaces/BUCKINGHAMSHIRE?search_id=testSearchSessionId"
      icon={
        <ResultAvatar
          appearance="square"
          borderColor="transparent"
          size="xsmall"
          src="http://lorempixel.com/640/480/cats"
        />
      }
      key="spaces_space-BUCKINGHAMSHIRE"
      screen="preQuerySearchResults"
      title="McLaughlin, Schowalter and Weissnat"
    />
    <ProductSearchResult
      analyticContext={
        Object {
          "containerId": null,
          "globalIndex": 4,
          "indexWithinSection": 1,
          "isCached": true,
          "sectionId": "confluence-space",
          "sectionIndex": 1,
        }
      }
      href="http://baseUrl/spaces/REDUNDANT?search_id=testSearchSessionId"
      icon={
        <ResultAvatar
          appearance="square"
          borderColor="transparent"
          size="xsmall"
          src="http://lorempixel.com/640/480/cats"
        />
      }
      key="spaces_space-REDUNDANT"
      screen="preQuerySearchResults"
      title="Hintz, O'Conner and Johns"
    />
    <ProductSearchResult
      analyticContext={
        Object {
          "containerId": null,
          "globalIndex": 5,
          "indexWithinSection": 2,
          "isCached": true,
          "sectionId": "confluence-space",
          "sectionIndex": 1,
        }
      }
      href="http://baseUrl/spaces/BUCKINGHAMSHIRE?search_id=testSearchSessionId"
      icon={
        <ResultAvatar
          appearance="square"
          borderColor="transparent"
          size="xsmall"
          src="http://lorempixel.com/640/480/cats"
        />
      }
      key="spaces_space-BUCKINGHAMSHIRE"
      screen="preQuerySearchResults"
      title="Mayer, Cole and Konopelski"
    />
  </SearchResultSection>
  <SearchResultSection
    title={
      <FormattedMessage
        defaultMessage="Recently worked with"
        description=""
        id="product_search_dialog.recently_worked_with_section_heading"
        values={Object {}}
      />
    }
  >
    <ProductSearchResult
      analyticContext={
        Object {
          "containerId": null,
          "globalIndex": 6,
          "indexWithinSection": 0,
          "isCached": true,
          "sectionId": "confluence-people",
          "sectionIndex": 2,
        }
      }
      containerDetail="@nickname-w9fiy"
      href="/people/928a0316-860c-406e-a101-ad5580c9854a?search_id=testSearchSessionId"
      icon={
        <ResultAvatar
          appearance="circle"
          borderColor="transparent"
          size="xsmall"
          src="http://lorempixel.com/640/480/people"
        />
      }
      key="people_people-928a0316-860c-406e-a101-ad5580c9854a"
      screen="preQuerySearchResults"
      title="Maude Wyman"
    />
    <ProductSearchResult
      analyticContext={
        Object {
          "containerId": null,
          "globalIndex": 7,
          "indexWithinSection": 1,
          "isCached": true,
          "sectionId": "confluence-people",
          "sectionIndex": 2,
        }
      }
      containerDetail="@nickname-wr8jn"
      href="/people/9796f716-2932-4252-9eb1-9c9de4c75c82?search_id=testSearchSessionId"
      icon={
        <ResultAvatar
          appearance="circle"
          borderColor="transparent"
          size="xsmall"
          src="http://lorempixel.com/640/480/people"
        />
      }
      key="people_people-9796f716-2932-4252-9eb1-9c9de4c75c82"
      screen="preQuerySearchResults"
      title="Jaquelin Schaden MD"
    />
    <ProductSearchResult
      analyticContext={
        Object {
          "containerId": null,
          "globalIndex": 8,
          "indexWithinSection": 2,
          "isCached": true,
          "sectionId": "confluence-people",
          "sectionIndex": 2,
        }
      }
      containerDetail="@nickname-83nid"
      href="/people/e5515743-7ce2-4045-a910-6485b4a3dfc8?search_id=testSearchSessionId"
      icon={
        <ResultAvatar
          appearance="circle"
          borderColor="transparent"
          size="xsmall"
          src="http://lorempixel.com/640/480/people"
        />
      }
      key="people_people-e5515743-7ce2-4045-a910-6485b4a3dfc8"
      screen="preQuerySearchResults"
      title="Chesley Bode"
    />
    <ConfluenceSearchResultSectionLink
      href="/wiki/people"
      id="people-search-link"
    >
      <span
        onClick={[Function]}
        role="none"
      >
        <FormattedMessage
          defaultMessage="Search people"
          description=""
          id="product_search_dialog.search_people_link"
          values={Object {}}
        />
      </span>
    </ConfluenceSearchResultSectionLink>
  </SearchResultSection>
  <WithAnalyticsEvents(WrappedComponent)
    isLoading={false}
    isPreQuery={true}
    resultCount={9}
    results={
      Array [
        Object {
          "results": Array [
            Object {
              "containerId": "UNAVAILABLE",
              "isRecentResult": false,
              "resultContentId": "a430a73d-9e9b-4cce-86a5-1618965ffea6",
              "resultType": "confluence-page",
            },
            Object {
              "containerId": "UNAVAILABLE",
              "isRecentResult": false,
              "resultContentId": "eec8184c-94bf-478c-9398-f8e79c41fa45",
              "resultType": "confluence-attachment",
            },
            Object {
              "containerId": "UNAVAILABLE",
              "isRecentResult": false,
              "resultContentId": "0e05cef4-042c-4adc-ad48-06618d01ced8",
              "resultType": "confluence-page",
            },
          ],
          "sectionId": "confluence-item",
        },
        Object {
          "results": Array [
            Object {
              "containerId": "UNAVAILABLE",
              "isRecentResult": true,
              "resultContentId": "space-BUCKINGHAMSHIRE",
              "resultType": "confluence-space",
            },
            Object {
              "containerId": "UNAVAILABLE",
              "isRecentResult": true,
              "resultContentId": "space-REDUNDANT",
              "resultType": "confluence-space",
            },
            Object {
              "containerId": "UNAVAILABLE",
              "isRecentResult": true,
              "resultContentId": "space-BUCKINGHAMSHIRE",
              "resultType": "confluence-space",
            },
          ],
          "sectionId": "confluence-space",
        },
        Object {
          "results": Array [
            Object {
              "containerId": "UNAVAILABLE",
              "isRecentResult": true,
              "resultContentId": "people-928a0316-860c-406e-a101-ad5580c9854a",
              "resultType": "person",
            },
            Object {
              "containerId": "UNAVAILABLE",
              "isRecentResult": true,
              "resultContentId": "people-9796f716-2932-4252-9eb1-9c9de4c75c82",
              "resultType": "person",
            },
            Object {
              "containerId": "UNAVAILABLE",
              "isRecentResult": true,
              "resultContentId": "people-e5515743-7ce2-4045-a910-6485b4a3dfc8",
              "resultType": "person",
            },
          ],
          "sectionId": "confluence-people",
        },
      ]
    }
    sectionCount={3}
  />
</span>
`;
